### DISCUSSION TITLE
Prompt Expansion in metadata

### DISCUSSION THREAD

**Discussion by barepixels** [MoonRide303/Fooocus-MRE#86] [General]
Is there a way to record such info in metadata?

[Prompt Expansion] New suffix: extremely detailed digital painting, in the style of fenghua zhong and ruan jia and jeremy lipking and peter mohrbacher, mystical colors, rim light, beautiful lighting, 8 k, stunning scene, raytracing, octane, trending on artstation

**Comment by 3Diva** [comment]
I'm not sure I understand the question (probably not lol) but if you're only asking about how to add a new "style" to the styles menu, you can add the prompt for the style in JSON format and put the JSON in the "sdxl_styles" folder. 

For instance the above prompt could be something like: 
`[
	{
		"name": "Magical Painting",
		"prompt": "extremely detailed digital painting, in the style of fenghua zhong and ruan jia and jeremy lipking and peter mohrbacher, mystical colors, rim light, beautiful lighting, 8 k, stunning scene, raytracing, octane, trending on artstation",
		"negative_prompt": "ugly, deformed, noisy, low poly, blurry"
	}
]`

You could save that as a JSON file and add it to your sdxl_styles folder and that should add a "Magical Painting" option to the styles menu.

**Comment by barepixels** [comment]
When you use the new Prompt expansion feature, Fooocus will add extra hidden prompt (Chat GPT 2.0 generated).  You can see this is the CMD console.

GPT2-based https://github.com/lllyasviel/Fooocus/discussions/117#raw (similar to Midjourney's hidden pre-processsing and "raw" mode, or the LeonardoAI's Prompt Magic).
https://github.com/lllyasviel/Fooocus

**Reply by 3Diva** [reply]
Oh I see, and you're wanting to see that hidden prompt also saved with the image like the text prompt. I think that would be a cool idea. It would be great if the entire prompt is saved with the image, even the Prompt Expansion hidden prompt.
